test/github-issues/11588/issue-11588.ts

```diff
@@ -0,0 +1,126 @@
+import "reflect-metadata"
+import { expect } from "chai"
+import { DataSource } from "../../../src/data-source/DataSource"
+import {
+    closeTestingConnections,
+    createTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { User } from "./entity/User"
+
+describe("github issues > #11588 Fine Grained Control over Update fields on Upsert", () => {
```

Comment:
We are moving away from creating tests for specific issues. Instead, please integrate tests for this feature into the relevant test suite(s) for EntityManager.